exercises/qos/README.md

```diff
@@ -0,0 +1,166 @@
+# Implementing QOS
+
+## Introduction
+
+The objective of this tutorial is to extend basic L3 forwarding with
+an implementation of Quality of Service (QOS) using Differentiated Services.
+
+Diffserv is simple and scalable. It classifies and manages network traffic and provides QOS on modern IP networks.
+
+As before, we have already defined the control plane rules for
+routing, so you only need to implement the data plane logic of your P4
+program.
+
+> **Spoiler alert:** There is a reference solution in the `solution`
+> sub-directory. Feel free to compare your implementation to the reference.
+
+## Step 1: Run the (incomplete) starter code
+
+The directory with this README also contains a skeleton P4 program,
+`qos.p4`, which initially implements L3 forwarding. Your job (in the
+next step) will be to extend it to properly append set the `diffserv` bits.
```

Comment:
Not sure of the intended meaning of this sentence.  Perhaps the intent is "will be to extend it to properly set the `diffuser` bits.", removing the word "append"?

Reply:
Done